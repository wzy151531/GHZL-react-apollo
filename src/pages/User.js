import React from 'react';
import { gql } from 'apollo-boost';
import { Query } from 'react-apollo';
import { Table } from 'antd';

const GET_RATES = gql`
  query GetRates{
    rates(currency: "USD") {
      currency, rate
    }
  }
`;

export const User = () => {
  return (
    <Query query={GET_RATES}>
      {({ loading, data }) => {
        const columns = [
          {
            key: 'currency',
            title: 'Currency',
            dataIndex: 'currency',
          },
          {
            key: 'rate',
            title: 'Rate',
            dataIndex: 'rate',
          },
        ];

        if (data.rates) {
          data.rates.forEach((n, i) => {
            n.key = i;
          });
        }
        const dataSource = data.rates || [];

        return (
          <div style={{ padding: '32px' }}>
            <Table
              size="small"
              loading={loading}
              dataSource={dataSource}
              columns={columns}
            />
          </div>
        );
      }}
    </Query>
  );
};