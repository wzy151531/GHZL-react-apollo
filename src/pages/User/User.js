import React from 'react';
import { gql } from 'apollo-boost';
import { Query } from 'react-apollo';
import { Table } from 'antd';
import { Link } from 'react-router-dom';

const GET_BOOKS = gql`
  query {
    books {
      title, author, id,
    }
  }
`;

export const User = () => {
  return (
    <Query query={GET_BOOKS}>
      {({ loading, data }) => {
        const columns = [
          {
            key: 'id',
            title: 'ID',
            dataIndex: 'id',
            render: id => (
              <Link to={`/user/${id}`} style={{ color: 'rgba(0, 0, 0, 0.65)' }}>{id}</Link>
            ),
          },
          {
            key: 'title',
            title: 'Title',
            dataIndex: 'title',
            width: 500,
          },
          {
            key: 'author',
            title: 'Author',
            dataIndex: 'author',
          },
        ];

        if (data.books) {
          data.books.forEach((n, i) => {
            n.key = i;
          });
        }
        const dataSource = data.books || [];

        return (
          <div style={{ padding: '32px' }}>
            <Table
              size="small"
              loading={loading}
              dataSource={dataSource}
              columns={columns}
            />
          </div>
        );
      }}
    </Query>
  );
};