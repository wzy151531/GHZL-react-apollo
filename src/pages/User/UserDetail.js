import React from 'react';

export const UserDetail = props => {
  return (
    <div>{`${props.match.params.id}'s detail`}</div>
  );
};