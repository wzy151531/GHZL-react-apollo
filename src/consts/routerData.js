import { Test } from '../pages/Test';
import { User } from '../pages/User';
import { Private } from '../pages/Private';
import { TestSub } from '../pages/TestSub';

export const routerData = [
  {
    path: '/',
    component: null,
    key: '2',
    authority: ['admin', 'sub-admin'],
    text: '',
    hideInMenu: true,
    breadcrumb: 'INDEX',
    alias: 'USER',
    intro: 'USER-INTRO',
  },
  {
    path: '/test',
    component: Test,
    key: 'sub1',
    authority: ['admin', 'sub-admin'],
    text: 'TEST',
    icon: 'user',
    breadcrumb: null,
    sub: [
      {
        path: '/test/sub',
        component: TestSub,
        key: '1',
        authority: ['admin'],
        text: 'TEST-SUB',
        breadcrumb: 'TEST-SUB',
        alias: 'TEST-SUB',
        intro: 'TEST-SUB-INTRO',
      },
    ],
  },
  {
    path: '/user',
    component: User,
    key: '2',
    authority: ['admin', 'sub-admin'],
    text: 'USER',
    icon: 'user',
    breadcrumb: 'USER',
    alias: 'USER',
    intro: 'USER-INTRO',
  },
  {
    path: '/private',
    component: Private,
    key: '3',
    authority: ['admin'],
    text: 'PRIVATE',
    icon: 'user',
    breadcrumb: 'PRIVATE',
    alias: 'PRIVATE',
    intro: 'PRIVATE-INTRO',
  },
];
