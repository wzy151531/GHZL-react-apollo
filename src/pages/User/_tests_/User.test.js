import React from 'react';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from '@apollo/react-hooks';
import User from '../User';
import renderer from 'react-test-renderer';
import { uri } from '../../../consts/apolloUri';

const client = new ApolloClient({
  uri: uri,
});

test('User should render correctly', () => {
  const component = renderer.create(
    <ApolloProvider client={client}><User/></ApolloProvider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});