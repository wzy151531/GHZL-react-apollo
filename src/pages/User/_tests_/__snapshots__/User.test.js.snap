// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render correctly 1`] = `
<div
  style={
    Object {
      "padding": "32px",
    }
  }
>
  <div
    className="ant-table-wrapper"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div>
        <div
          className="ant-spin ant-spin-spinning ant-table-without-pagination ant-table-spin-holder"
        >
          <span
            className="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <div
        className="ant-spin-container ant-spin-blur"
      >
        <div
          className="ant-table ant-table-small ant-table-empty ant-table-scroll-position-left"
          style={Object {}}
        >
          <div
            className="ant-table-content"
          >
            <div
              className="ant-table-body"
              onScroll={[Function]}
              onWheel={[Function]}
              style={Object {}}
            >
              <table
                className=""
                style={Object {}}
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": 500,
                        "width": 500,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            ID
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            Title
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            Author
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                />
              </table>
            </div>
            <div
              className="ant-table-placeholder"
            >
              <div
                className="ant-empty ant-empty-normal"
              >
                <div
                  className="ant-empty-image"
                >
                  <svg
                    height="41"
                    viewBox="0 0 64 41"
                    width="64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fillRule="evenodd"
                      transform="translate(0 1)"
                    >
                      <ellipse
                        cx="32"
                        cy="33"
                        fill="#F5F5F5"
                        rx="32"
                        ry="7"
                      />
                      <g
                        fillRule="nonzero"
                        stroke="#D9D9D9"
                      >
                        <path
                          d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                        />
                        <path
                          d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                          fill="#FAFAFA"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
                <p
                  className="ant-empty-description"
                >
                  No Data
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
